libimage/runtime.go

```diff
@@ -408,22 +409,17 @@ func (r *Runtime) lookupImageInDigestsAndRepoTags(name string, options *LookupIm
 				}
 			}
 		}
-
 		return nil, "", errors.Wrap(storage.ErrImageUnknown, name)
 	}
 
-	// Podman compat: if we're looking for a short name but couldn't
-	// resolve it via the registries.conf dance, we need to look at *all*
-	// images and check if the name we're looking for matches a repo tag.
-	// Split the name into a repo/tag pair
-	split := strings.SplitN(name, ":", 2)
-	repo := split[0]
-	tag := ""
-	if len(split) == 2 {
-		tag = split[1]
+	if !shortnames.IsShortName(name) {
+		return nil, "", errors.Wrap(storage.ErrImageUnknown, name)
 	}
+
+	named = reference.TagNameOnly(named) // Make sure to add ":latest" if needed
+	namedTagged, _ := named.(reference.NamedTagged)
```

Comment:
This shouldn’t just ignore the second value; the loop is invalid if the conversion fails.

I know, the `isDigested` code above has already exited for values that would fail this conversion, but that’s non-obvious, worth at least a comment here. But I think the https://github.com/containers/common/pull/836#discussion_r754254334 structure would make this moot, with no dependencies between the two approaches.

Reply:
> I prefer the structure as is though.

OK.